### PR TITLE
fix reloading problem

### PR DESCRIPTION
**Pull request by sanaAB**

### CODE REVIEW

**Review by sanaAB — COMMENTED** [review]
Changes reviewed

File: src/components/Recipe/Recipe.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ export default function Recipe(props) {
         .map((recipe) => (
           <li className="recipes_list">
           <Link to={`/${category}/${recipe.fields.slug}`}>
-            <h4>{recipe.fields.name}</h4>
+            <h2 className="hhh">{recipe.fields.name}</h2>
```
Comment (gsambrotta) [inline src/components/Recipe/Recipe.js:17]: `hhh` is a very creative name for a class :P 

Please gives something more meaningful, like title, subtitle, h2 or similar.

File: src/components/RecipePage/RecipePage.js [hunk 2]
```diff
@@ -15,9 +15,9 @@ const useStyles = makeStyles(() => ({
   },
 }));
 
-export default function RecipePage(props) {
+export default function RecipePage(props, {isLoading}) {
   const classes = useStyles();
-
+  console.log(props);
```
Comment (gsambrotta) [inline src/components/RecipePage/RecipePage.js:20]: to be removed

File: src/components/RecipePage/RecipePage.js [hunk 3]
```diff
@@ -27,34 +27,47 @@ export default function RecipePage(props) {
         return (singleRecipe = recipe.fields);
       }
     });
-  console.log(singleRecipe);
-  const postDescription = marked(singleRecipe.description);
+  console.log(props.isLoading);
```
Comment (gsambrotta) [inline src/components/RecipePage/RecipePage.js:30]: ditto

File: src/components/RecipePage/RecipePage.js [hunk 4]
```diff
@@ -15,9 +15,9 @@ const useStyles = makeStyles(() => ({
   },
 }));
 
-export default function RecipePage(props) {
+export default function RecipePage(props, {isLoading}) {
```
Comment (gsambrotta) [inline src/components/RecipePage/RecipePage.js:18]: This is not the best way to pass down props.
You can either pass the whole `props` object:
i.e. `... RecipiePage(props)` and then grab your props one by one
like `props.RecipieArray`, `props.isLoading`

or you can deconstruct your props object:
i.e. `...RecipePage({ description, name, isLoading, ... })`
and use them directly in your code.

We have done many other times, when in doubt please check your older projects or other components:

File: src/components/RecipePage/RecipePage.js [hunk 5]
```diff
@@ -27,34 +27,47 @@ export default function RecipePage(props) {
...
-              dangerouslySetInnerHTML={{ __html: postDescription }}
-            />
-          }
-        </div>
+      { isLoading ? (
+      <div className="load__style">
+        <h4>Loading recipe...</h4>
+        <i className="fa fa-spinner fa-pulse fa-3x fa-fw"></i>
+      </div>
+    ) : singleRecipe ? (
```
Comment (gsambrotta) [inline src/components/RecipePage/RecipePage.js:43]: why you have another conditional here? 
If you get till here, you are already sure you have singleRecipie.
Either you check if isLoading is true or you check if singleRecipe is not undefined.

File: src/App.js [hunk 6]
```diff
@@ -49,13 +49,13 @@ function App() {
               <Recipe recipesArray={recipesArray} />
             </Route>
             <Route exact path="/breakfast/:slug">
```
Comment (gsambrotta) [inline src/App.js:51]: You don't need to define all your categories manually.
You can just write `<Route path='/:category/:slug'>`

**Review by gsambrotta — COMMENTED** [review]
Good job fixing the reloading with the isLoading!
I left you some comments to improve your code, please have a look

**Pull request merged**